feature_engine/prediction/target_mean_prediction.py

```diff
@@ -0,0 +1,244 @@
+        self.variables_categorical_ = list(X.select_dtypes(include="object").columns)
+        self.variables_numerical_ = list(X.select_dtypes(include="number").columns)
+
+        # transform categorical and numerical variables, where appropriate
+        if self.variables_categorical_ and self.variables_numerical_:
+            _pipeline = self._make_combined_pipeline()
+
+        elif self.variables_categorical_:
+            _pipeline = self._make_categorical_pipeline()
+
+        else:
+            _pipeline = self._make_numerical_pipeline()
+
+
+        # check if df contains na
+        _check_contains_na(X, self.variables_)
+
+        self.n_features_in_ = X.shape[1]
+
+        return self
+
+    def predict(self, X: pd.DataFrame, variable_name: str) -> pd.Series:
+        """
+
+        Parameters
+        ----------
+        X : pandas dataframe of shape = [n_samples, ]
+            The input series which must have the same name as one of the features in the
+            dataframe that was used to fit the predictor.
+
+        variable_name: str
+            The variable the method should transform and return.
+
+        Return
+        -------
+        X_prediction: pandas series of shape = [n_samples, ]
+            Values are the mean values associated with the corresponding encoded or discretised bin
+
+        """
+        # NOTES:
+        # - X needs to be a dataframe to be compatible w/ the BaseEncoder()
+        # - X needs to match the shape of the dataframe used in fit()
+
+        # check if is pandas series w/ a name that matches
+        #
+        # if not isinstance(X, pd.Series):
+        #     raise TypeError("fit() method only accepts pandas series.")
+
+        if variable_name not in (self.variables_):
+            raise ValueError("Series name does not match the dataframe features that were used to "
+                             "fit the predictor. The variables that were fitted to the predictor: "
+                             f"{self.varialbes_}.")
+
+        # use fitted encoder to return the corresponding mean for each category
+        if variable_name in self.variables_categorical_:
+            X_prediction = self.encoder.transform(X)
+
+        # transform the numerical to the appropriate discretised bin.
+        # replace the bin index w/ the corresponding mean value.
+        else:
+            X_transformed = self.discretiser.transform(X)
+            X_prediction = X_transformed.apply(lambda bin_idx: self.disc_mean_dict_[bin_idx])
+
+        return X_prediction[variable_name]
+
+    def _make_numerical_pipeline(self):
+
+        if self.strategy == "equal-width":
+            discretiser = EqualWidthDiscretiser(
+                bins=self.bins, variables=self.variables_numerical_, return_object=True
+            )
+        else:
+            discretiser = EqualFrequencyDiscretiser(
+                q=self.bins, variables=self.variables_numerical_, return_object=True
+            )
+
```

Comment:
I would capture lines 196 to 203 in a new method() `make_discretiser()`, that returns the discretizer. So that we can use this method instead of repeating all of this code in the _make_numerical_pipeline() and _make_combined_pipeline()